admin/class-yoast-form.php

```diff
@@ -0,0 +1,419 @@
+
+		echo '<label class="textinput" for="' . esc_attr( $var ) . '">' . esc_html( $label ) . ':</label><textarea cols="' . esc_attr( $attr['cols'] ) . '" rows="' . esc_attr( $attr['rows'] ) . '" class="textinput ' . esc_attr( $attr['class'] ) . '" id="' . esc_attr( $var ) . '" name="' . esc_attr( $this->option ) . '[' . esc_attr( $var ) . ']">' . esc_textarea( $val ) . '</textarea>' . '<br class="clear" />';
+	}
+
+	/**
+	 * Create a hidden input field.
+	 *
+	 * @param string $var The variable within the option to create the hidden input for.
+	 */
+	public function hidden( $var ) {
+		$val = ( isset( $this->options[ $var ] ) ) ? $this->options[ $var ] : '';
+		if ( is_bool( $val ) ) {
+			$val = ( $val === true ) ? 'true' : 'false';
+		}
+
+		echo '<input type="hidden" id="hidden_' . esc_attr( $var ) . '" name="' . esc_attr( $this->option ) . '[' . esc_attr( $var ) . ']" value="' . esc_attr( $val ) . '"/>';
+	}
+
+	/**
+	 * Create a Select Box.
+	 *
+	 * @param string $var    The variable within the option to create the select for.
+	 * @param string $label  The label to show for the variable.
+	 * @param array  $values The select options to choose from.
+	 */
+	public function select( $var, $label, $values ) {
+		if ( ! is_array( $values ) || $values === array() ) {
+			return;
+		}
+		$val = ( isset( $this->options[ $var ] ) ) ? $this->options[ $var ] : '';
+
+		$output = '<label class="select" for="' . esc_attr( $var ) . '">' . $label . ':</label>';
+		$output .= '<select class="select" name="' . esc_attr( $this->option ) . '[' . esc_attr( $var ) . ']" id="' . esc_attr( $var ) . '">';
+
+		foreach ( $values as $value => $label ) {
+			if ( ! empty( $label ) ) {
+				$output .= '<option value="' . esc_attr( $value ) . '"' . selected( $val, $value, false ) . '>' . $label . '</option>';
+			}
+		}
+		$output .= '</select>';
+
+		echo $output . '<br class="clear"/>';
+	}
+
+	/**
+	 * Create a File upload field.
+	 *
+	 * @param string $var   The variable within the option to create the file upload field for.
+	 * @param string $label The label to show for the variable.
+	 */
+	public function file_upload( $var, $label ) {
+		$val = '';
+		if ( isset( $this->options[ $var ] ) && is_array( $this->options[ $var ] ) ) {
+			$val = $this->options[ $var ]['url'];
+		}
+
+		$var_esc = esc_attr( $var );
+		$output  = '<label class="select" for="' . $var_esc . '">' . esc_html( $label ) . ':</label>';
+		$output .= '<input type="file" value="' . esc_attr( $val ) . '" class="textinput" name="' . esc_attr( $this->option ) . '[' . $var_esc . ']" id="' . $var_esc . '"/>';
+
+		// Need to save separate array items in hidden inputs, because empty file inputs type will be deleted by settings API.
+		if ( ! empty( $this->options[ $var ] ) ) {
+			$output .= '<input class="hidden" type="hidden" id="' . $var_esc . '_file" name="wpseo_local[' . $var_esc . '][file]" value="' . esc_attr( $this->options[ $var ]['file'] ) . '"/>';
```

Comment:
Why not using the hidden method. Only thing we can do is make an option attribute in method hidden named `$id`.